### PR TITLE
stat s3

### PR DESCRIPTION
**Pull request by nchinan**
.flatMap(fs.statStream) will throws an exception when an s3 bucket uri is supplied.  I have just set the path to '/' in this case for the check.

### CODE REVIEW

**Comment by robertstettner** [comment]
There is a better way. Change the condition here: https://github.com/nchinan/drone-cloudformation/blob/fd3ef84e465f383b044a5de9d3dda642a949f6e7/plugin.js#L104

**Comment by nchinan** [comment]
I prefer something like

```
const hl = require('highland');
const path = require('path');
let fs = hl.streamifyAll(require('fs'));


let resolveAbosolutePath = p => path.resolve(process.cwd(), p);


const p = ['./node_modules', 's3://mybucket/file.zip', 'https://s3.mybucket.com/file.zip'];

if (process.env.ADD_BAD_FILE) {
        p.push('./asdfasdfsda');
}

let s = hl(p)
        .reject(x => /s3:\/\//.test(x) || /https:\/\//.test(x))
        .map(resolveAbosolutePath)
        .flatMap(fs.statStream)
        .collect()
        .toCallback(err => {
                if (err) {
                        console.error(err);
                        return process.exit(1);
                }
                process.exit(0);
        });


```

File: plugin.js [hunk 1]
```diff
@@ -102,7 +102,9 @@ let validate = function (envs) {
         .flatMap(env => hl.of(validateConfig(env)))
         .flatMap(env =>{
             if (env.PLUGIN_MODE !== 'delete') {
-                return hl.of(resolveAbsolutePath(env.PLUGIN_TEMPLATE))
+                return hl([env.PLUGIN_TEMPLATE])
```
Comment (robertstettner) [inline plugin.js]: Please use the `of` method instead of encapsulating the template in an array.

**Review by robertstettner — CHANGES_REQUESTED** [review]
Could you add a unit test for this here: https://github.com/robertstettner/drone-cloudformation/blob/d5da579286270adf686532ac91309c1a66d6ffea/test/unit.spec.js#L549

**Comment by robertstettner** [comment]
Please remove the `.swp` file, and while you there, could you please add that extension to the `.gitignore` file?

**Review by robertstettner — APPROVED** [review]

**Pull request merged**